src/jobs/ega/services/permissions.ts

```diff
@@ -113,10 +193,16 @@ export const processPermissionsForApprovedUsers = async (
             (perm) => perm.dataset_accession_id,
           );
           const datasetsRequiringPermissions = datasets.map((dataset) => dataset.accession_id);
+          // if a dataset is removed from the incoming datasets list argument, i.e. the user has more dataset permissions than the incoming list,
```

Comment:
just left this comment here, to remind me (and perhaps others) how this call to `difference()` should behave. Not sure how likely it is that a dataset would be removed!